scipy/_lib/_array_api.py

```diff
@@ -528,11 +528,11 @@ def xp_sign(x: Array, /, *, xp: ModuleType | None = None) -> Array:
     xp = array_namespace(x) if xp is None else xp
     if is_numpy(xp):  # only NumPy implements the special cases correctly
         return xp.sign(x)
-    sign = xp.full_like(x, xp.asarray(xp.nan))
+    sign = xp.zeros_like(x)
```

Comment:
This didn't work for integer dtypes. Instead of starting with NaNs and replacing with zeros as necessary, do the opposite.